controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -101,14 +103,19 @@
     // Send to the client server traces on error message replies.
     private final boolean replyWithStackTraceOnError;
 
+    private final boolean isInternalWritesWithReadPermEnabled;
```

Comment:
True, the current configuration is broadly applied - all internal streams can be written to with "read" permissions if this config is true, otherwise, they'll require "write" permissions. 

Making it tunable at a stream-level can provide more flexibility. However, it could also hurt usability - presumably, users shall have to make another explicit choice and it wouldn't be easy for them to determine what to do. And, they'd have to understand a bit more about internal workings for being able to decide what to do.

I'd say we should keep the coarse-grained config as of now, and introduce more granular config if need be at a later point. What do you say?

Reply:
IMHO, it is really a system wide setting that the admin/operator installing Pravega decides.